kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerOrchestrationService.java

```diff
@@ -41,7 +41,37 @@ public interface ContainerOrchestrationService {
      * @return a list of {@link ContainerInstanceDescriptor}s representing the available containers
      */
     public List<ContainerInstanceDescriptor> listContainerDescriptors();
-
+    
+    /**
+     * @since 2.4
+     * Lists all available images, returning the corresponding {@link ImageInstanceDescriptor}s
+     *
+     * @return a list of {@link ContainerInstanceDescriptor}s representing the available containers
+     */
+    public List<ImageInstanceDescriptor> listImageInstanceDescriptor();
+    
+    /**
+     * @since 2.4
+     * Deletes Image from container engine. Takes ImageID String as parameter. Will throw an error if image is being used by a container.
+     *
+     * @return void
+     * @throws KuraException 
+     */
+    public void deleteImage(String imageId) throws KuraException;
+    
+    /**
+     * @since 2.4
+     * Allows to pull the required image, using the specified tag and credentials. The image will be downloaded
+     * respecting the configured timeout in seconds.
+     *
+     * @param imageConfig
+     *            a ImageConfiguration object which contains info such as image name, tag, pull timeout in secconds and registry credentials.
+     * @throws KuraException
+     *             if the pull operation fails
+     * @throws InterruptedException
+     */
```

Comment:
/**
     * Allows to pull the required image, using the specified tag and credentials. The image will be downloaded
     * respecting the configured timeout in seconds.
     *
     * @param imageConfig
     *            a ImageConfiguration object which contains info such as image name, tag, pull timeout in seconds and
     *            registry credentials.
     * @throws KuraException
     *             if the pull operation fails
     * @throws InterruptedException
     * `@since 2.4`
     */